### PR TITLE
[JENKINS-44563] Use one-column layout for REST API page

### PR DESCRIPTION
**Pull request by recena**
# Description

See [JENKINS-44563](https://issues.jenkins-ci.org/browse/JENKINS-44563).

## Before
jenkins-44564_before

## After
jenkins-44563_after2

### Changelog entries

Proposed changelog entries:

* REST API page is using one-column layout

### Submitter checklist

- [X] JIRA issue is well described
- [X] Link to JIRA ticket in description, if appropriate

### Desired reviewers

@jenkinsci/code-reviewers 
@reviewbybees

### CODE REVIEW

**Review by oleg-nenashev — COMMENTED** [review]
There are many unrelated changes, and I am not sure it is a good idea to use `code` tags for non-code text. Needs a separate discussion.

Regarding the type change itself, 👍 from me. Ideally in the column there could be ToC or sidepanel, but it is a separate story.

File: core/src/main/resources/hudson/model/Api/index.jelly
Comment (recena) [inline core/src/main/resources/hudson/model/Api/index.jelly]: @oleg-nenashev We could consider this pieces of text as _code_ https://developer.mozilla.org/en/docs/Web/HTML/Element/code

File: core/src/main/resources/hudson/model/Api/index.jelly
Comment (recena) [inline core/src/main/resources/hudson/model/Api/index.jelly]: @oleg-nenashev Maybe we need a special style for reserved words.

**Comment by recena** [comment]
@oleg-nenashev 

> Regarding the type change itself, 👍 from me. Ideally in the column there could be ToC or sidepanel, but it is a separate story.

Sure. With this layout we could include user-friendly documentation. More space, more flexibiliy, etc...

File: core/src/main/resources/hudson/model/Api/index.jelly
Comment (jglick) [inline core/src/main/resources/hudson/model/Api/index.jelly]: Why this formatting change?

File: core/src/main/resources/hudson/model/Api/index.jelly
Comment (jglick) [inline core/src/main/resources/hudson/model/Api/index.jelly]: This is the only part of the diff actually implementing the stated change IIUC.

**Review by jglick — APPROVED** [review]

File: core/src/main/resources/hudson/model/Job/_api.jelly
Comment (jglick) [inline core/src/main/resources/hudson/model/Job/_api.jelly]: Ditto—gratuitous diff AFAICT.

File: core/src/main/resources/hudson/model/Api/index.jelly
Comment (recena) [inline core/src/main/resources/hudson/model/Api/index.jelly]: readability?

File: core/src/main/resources/hudson/model/Job/_api.jelly
Comment (recena) [inline core/src/main/resources/hudson/model/Job/_api.jelly]: @jglick Do you prefer a separate PR for applying some of style-format?

File: core/src/main/resources/hudson/model/Api/index.jelly
Comment (jglick) [inline core/src/main/resources/hudson/model/Api/index.jelly]: Just please avoid touching unrelated lines of code when filing a PR. The longer the diff, the harder it is to review, thus the less likely it is to be merged (and the more likely your change is to cause problems for merging other changes, cherry-picking fixes to LTS, `blame`ing, etc.).

Comment (recena) [inline core/src/main/resources/hudson/model/Api/index.jelly]: @jglick I don't believe this is such case. This file has had (and has) very slow activity, I don't see the problem. At any case, changes related with the UI, are susceptible of resulting complex for reviewing, for that reason, I try to provide screenshots and manual tests.

Comment (recena) [inline core/src/main/resources/hudson/model/Api/index.jelly]: And yes, I believe this project needs to review the code-formatting of some files 😄

Comment (daniel-beck) [inline core/src/main/resources/hudson/model/Api/index.jelly]: > I believe this project needs to review the code-formatting of some files

Including unrelated changes with a PR is not the way to do that.

Comment (recena) [inline core/src/main/resources/hudson/model/Api/index.jelly]: @daniel-beck @jglick These small style-format changes are blocking this PR?

Comment (oleg-nenashev) [inline core/src/main/resources/hudson/model/Api/index.jelly]: @recena Integrating obviously unrelated changes into a single PR slows down review and merge of the real fix. It is convenient neither for reviews nor for the committer himself. Any reviewer may **legitimately** block the PR due to unrelated changes, so by adding them you accept the risk of such blockage. Just IMHO.

**Comment by kuisathaverat** [comment]
nice improvement 👍

**Comment by recena** [comment]
@kuisathaverat Thanks for your feedback.

File: core/src/main/resources/hudson/model/Api/index.jelly
Comment (daniel-beck) [inline core/src/main/resources/hudson/model/Api/index.jelly]: @recena It's not a big deal if it happens occasionally. We also don't expect you to not do an obvious improvement near the code you need to change anyway (let's say "near" is "`git diff` doesn't skip unmodified lines in between" as a rule of thumb). But IIRC inclusion of unrelated changes, even in otherwise unchanged files, has been a recurring issue in PRs from you. We bring it up every time it happens, but you don't internalize our feedback. In fact, it looks deliberate, despite being asked not to do it.

What would you expect us to do in this situation?

Comment (recena) [inline core/src/main/resources/hudson/model/Api/index.jelly]: @daniel-beck Again. Is this blocking the PR merge? If so, I'll revert the changes.

File: core/src/main/resources/hudson/model/Api/index.jelly
Comment (recena) [inline core/src/main/resources/hudson/model/Api/index.jelly]: Removing tab characters...

**Review by jglick — APPROVED** [review]
FWIW I was not blocking the PR before—I approved it—but this is indeed better, thanks.

A PR replacing `tt` with `code` would be fine. I would generally object to a PR which just reformats code for fun, unless the original condition was truly unreadable (which is certainly not the case here).

**Comment by recena** [comment]
@jglick Thanks so much.

> I would generally object to a PR which just reformats code for fun, unless the original condition was truly unreadable (which is certainly not the case here).

I never do changes for fun 😄

**Review by daniel-beck — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]

**Comment by recena** [comment]
The PR builder is failing. Unrelated at all.

**Comment by oleg-nenashev** [comment]
@reviewbybees done I'd guess

**Pull request merged**
